Neeraj Job Posting Page Analytics Frontend : Create a horizontal bar graph to show hits/applications ratio

Neeraj-Kondaveeti opened the pull request:
# Description
Job Posting Page Analytics: Create a horizontal bar graph to show hits/applications ratio

<img width="695" height="468" alt="image" src="https://github.com/user-attachments/assets/ea0f7559-9e50-444d-9829-6a597204c26d" />

<img width="756" height="616" alt="image" src="https://github.com/user-attachments/assets/1f2c09a5-757c-4599-9ec1-c1c5e1a4d31d" />

<img width="697" height="231" alt="image" src="https://github.com/user-attachments/assets/c9d44dc9-0417-4b5f-b441-86300dacccc9" />


Include a toggle switch to switch between percentage and absolute values.

## Related PRS (if any):
This frontend PR is related to the https://github.com/OneCommunityGlobal/HGNRest/pull/1656 backend PR.

This PR incorporates updates based on the feedback from Old PR #3582 . Original reviewers are tagged for continuity to confirm that their earlier comments have been addressed.
…

## Main changes explained:
added two graphs for 1.Converted Applications Graph 2.Non-Converted Applications Graph
added necessary filters and toggle buttons
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to http://localhost:5173/job-analytics
6. verify graphs are shown with values/percentages with filters


## Screenshots or videos of changes:

<img width="2930" height="1722" alt="image" src="https://github.com/user-attachments/assets/84c88c71-8356-46da-826f-66988a7fa12e" />

<img width="2926" height="1728" alt="image" src="https://github.com/user-attachments/assets/1d7a9db4-ffda-4fb8-8e66-c13f4ed4ec85" />

PipKcK reviewed (changes requested):
I am unable to view data
<img width="1919" height="430" alt="image" src="https://github.com/user-attachments/assets/351d13a6-1ed6-4192-8b92-97e58f91d9ab" />
 FE and BE on the correct branch

harshavarma29 reviewed (changes requested):
I have tested the PR locally using both Admin and Test accounts, and it is working as expected. However, I have noticed the following UI issues.

1) Dark mode is not working as expected:
<img width="1882" height="469" alt="dark_mode_issue" src="https://github.com/user-attachments/assets/2c91a165-6327-41a0-ad3a-f659a06f79f5" />

2) It would be better to display names for the axis labels:
- X-axis: Show either **Percentage of Hits Converted to Applications** or **Actual number of applications** based on the toggle.
- Y-axis: Show **Job role**
<img width="1863" height="625" alt="x_y_axis_labels" src="https://github.com/user-attachments/assets/b664493f-0c6c-4ba5-aa0c-e32849004b00" />

Neeraj-Kondaveeti commented:
> I have tested the PR locally using both Admin and Test accounts, and it is working as expected. However, I have noticed the following UI issues.
> 
> 1. Dark mode is not working as expected:
> 
> <img alt="dark_mode_issue" width="1882" height="469" src="https://private-user-images.githubusercontent.com/46283874/481340059-2c91a165-6327-41a0-ad3a-f659a06f79f5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzNDM3OTAsIm5iZiI6MTc1NjM0MzQ5MCwicGF0aCI6Ii80NjI4Mzg3NC80ODEzNDAwNTktMmM5MWExNjUtNjMyNy00MWEwLWFkM2EtZjY1OWEwNmY3OWY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI4VDAxMTEzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMTA5MzVlODRlZWQ5YmJjYTliZGY2ODBhNzRkMjIzOWZlNjNlNTEzMDVhZDgxYTc4YTY0MGI4ODY3NWM2OTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NVyN-fWkOWGKkWvhm8r3uAlizScMnlgv3SXoL8SdOZc">
> 2. It would be better to display names for the axis labels:
> 
> * X-axis: Show either **Percentage of Hits Converted to Applications** or **Actual number of applications** based on the toggle.
> * Y-axis: Show **Job role**
> 
> <img alt="x_y_axis_labels" width="1863" height="625" src="https://private-user-images.githubusercontent.com/46283874/481340069-b664493f-0c6c-4ba5-aa0c-e32849004b00.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzNDM3OTAsIm5iZiI6MTc1NjM0MzQ5MCwicGF0aCI6Ii80NjI4Mzg3NC80ODEzNDAwNjktYjY2NDQ5M2YtMGM2Yy00YmE1LWFhMGMtZTMyODQ5MDA0YjAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI4VDAxMTEzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0MmM5MDkzNGQ3MmY0YWM3ZTZjZjhlMGZjZGU1YmI4YWRmOWRkNDAzNDEzZWEwYWRkMmZmYWNiMTg1OGY0NGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fHHMbfFoJxI563QGxurtv6JRFSY4tAb7qlESGaREeek">


Hi Harsha , 

I am not expected to work on dark mode based on the requirements given in the task description. Dark mode for this chart is not required as far as I know since it is not mentioned in the requirements which can be seen In the description section at the start of the page. Coming to your second point , percentage of hits or actual number of applications can be seen in the hover tooltip if you hover over each bar. And the Job Role is visible on the Y axis as seen in your screenshot. Let me know if you have any other concerns.

Neeraj-Kondaveeti commented:
> Hi @Neeraj-Kondaveeti,
> 
> Thank you for addressing the comments.
> 
> 1. Dark mode: Adding dark mode enhances the User Experience even though it is not mentioned in the requirements. It is more like making an additional contribution, improvement, or suggestion from our side.
> 2. Labels: Adding labels improves clarity. Adding labels for the X-axis and Y-axis is mentioned in the requirements. I have attached the screenshot for reference.
> 
> <img alt="graph-label-requirement" width="646" height="616" src="https://private-user-images.githubusercontent.com/46283874/483922709-c470b3c0-081f-4a71-a026-3114b2d0fc53.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc0NTg4NDcsIm5iZiI6MTc1NzQ1ODU0NywicGF0aCI6Ii80NjI4Mzg3NC80ODM5MjI3MDktYzQ3MGIzYzAtMDgxZi00YTcxLWEwMjYtMzExNGIyZDBmYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA5VDIyNTU0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkMTIyNGQwZTI3NTA5NThlMzA0MTQ5MGFlMzZjM2U5ZWE2NWMzNTU0Yjc3OGFkMGZkODEyZjAxMzhiMGViOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GeMnV9zGty6fBCx6hlDGKo3lh0a2VZ6WWmgDEYXD7Zw">


Hi Harsha , 

Hope you are doing well , thank you for the recommendations . I have made the requested changes , please test this out again and let me know if everything is or if anything needs to be taken care of. 

Thank You

harshavarma29 reviewed (changes requested):
I have tested the PR locally, and it is working as expected. The X and Y axis labels are displayed correctly. However, I noticed below issue in dark mode:

The dark theme colors do not match the recommended palettes from the [Dark Mode Implementation Guide](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI/edit?tab=t.0).

Neeraj-Kondaveeti commented:
> I have tested the PR locally, and it is working as expected. The X and Y axis labels are displayed correctly. However, I noticed below issue in dark mode:
> 
> The dark theme colors do not match the recommended palettes from the [Dark Mode Implementation Guide](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI/edit?tab=t.0). <img alt="dark_mode_not_working" width="1818" height="795" src="https://private-user-images.githubusercontent.com/46283874/489095572-49d93e0a-4ad0-473a-93e6-703a28acd439.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc5ODI5OTEsIm5iZiI6MTc1Nzk4MjY5MSwicGF0aCI6Ii80NjI4Mzg3NC80ODkwOTU1NzItNDlkOTNlMGEtNGFkMC00NzNhLTkzZTYtNzAzYTI4YWNkNDM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE2VDAwMzEzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiZGI4MzJjZTM1NTM2YzllOTk3YjQ1ZjYzOGEyY2YyMzYzYzUyYTY1NmQ0NWNhMTE3Yzg4ZjgwOWNkNTE4OGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OJXnwzU0tTAz7-tvGxQ9BmHBZBcgCC1mZ1KD1Ts_RYI">

Hi Harsha , thank you for pointing that out. I have made the changes such that the dark theme colors match the recommended palettes from the Dark Mode Implementation guide. Can you check once again and let me know if everything is working as expected.

Thank You

RitzzzZ2021 reviewed (dismissed):
Tested the chart, and it meets all the requirements outlined in the description. Toggling the box successfully switches between percentage and actual values, and the hover tooltips display the required additional information. Great work!

<img width="1468" height="674" alt="image" src="https://github.com/user-attachments/assets/e8bf62b2-2b0f-4e89-83ee-9475055b6891" />

<img width="1469" height="661" alt="image" src="https://github.com/user-attachments/assets/7fa39c07-490b-43b6-83e3-8be8ba4d5899" />

<img width="1464" height="666" alt="image" src="https://github.com/user-attachments/assets/be7bc693-c70a-42fb-9912-2cc24637fbe7" />

Suggestion: consider adjusting the scale of the second chart when displaying percentages, as the values are typically below 10 but the scale is set to 100.

Nhujarski reviewed (approved):
I reviewed the PR given the steps above. The graphs look great and function as expected. Good job!

<img width="1440" height="764" alt="pr 3889 1" src="https://github.com/user-attachments/assets/aec15b5b-1087-4202-b5b1-21233062b8b3" />
<img width="1440" height="764" alt="pr 3889 2" src="https://github.com/user-attachments/assets/902cf845-4f73-4026-8376-79cde8f088d8" />

one-community commented:
Thank you all, merging!
